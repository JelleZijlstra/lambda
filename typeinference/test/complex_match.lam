data t = C1 int | C2 bool bool | C3 (int * bool) in
let f = \t. match t with {
	| C1 3 -> print 3
	| C1 _ -> print 1
	| C2 true false -> print 2
	| C2 _ b -> print (if b then 4 else 5)
	| C3 (3, true) -> print 6
	| C3 (_, _) -> print 7
} in
(f (C1 3); # 3
f (C1 2); # 1
f (C2 true false); # 2
f (C2 false true); # 4
f (C2 false false); # 5
f (C3 (3, true)); # 6
f (C3 (4, true)); # 7
f)
