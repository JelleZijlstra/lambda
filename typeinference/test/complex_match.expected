%%stdout%%
3
1
2
4
5
6
7
Result: module
	let __result__ = \t. match t with C1 3 -> print 3 | C1 _ -> print 1 | C2 true false -> print 2 | C2 _ b -> print if b then 4 else 5 | C3 (3, true) -> print 6 | C3 (_, _) -> print 7
end
%%stderr%%
